DevilutionX: make script check for spawn.mpq

Peppersawce opened the pull request:
As requested in the [forums](https://discuss.haiku-os.org/t/devilutionx-testing-1-5-4-2/16638), a small change to the startup script so it accepts the game demo's data as valid for playing DevilutionX

humdingerb commented:
Would itbe possible to put the game "devilutionx" directly into system/apps/ and the "devilutionx.sh" script in /system/bin/ ? I think we should try to put things in the top folder when possible.

Also, can the binary be renamed "DevilutionX"?

Peppersawce commented:
> Would itbe possible to put the game "devilutionx" directly into system/apps/ and the "devilutionx.sh" script in /system/bin/ ? I think we should try to put things in the top folder when possible.
> 
> Also, can the binary be renamed "DevilutionX"?

Both can be done, at least for now, I have been asked however by the DevilutionX guys to try and get their assets packed using "smpq" (which has to be ported but they showed me a working script I can base a recipe off of), when that happens it's going to need its own folder in apps, but the script could stay in bin... should be an ok compromise?
After that's done, I could also try if symlinking the binary to the apps folder works.

> Oh, and do the settings need to be in a "diasurgical" subfolder?
It needlessly adds another level and finding your settings when you're looking for "devilution" gets harder.

That's the default naming for the settings folder but yeah sure, that can be patched out I think.

I'll try and tinker with these suggestions soon, I don't think I have the time for it today though. :)

humdingerb commented:
👍

Peppersawce commented:
Tried putting the binary into "apps" and it started segfaulting looking for its assets there, my guess is that the binary looks for its assets in this folder first then uses the ones I supplied with my old patch as fallback.

So I moved binary, script and assets into their own folder and symlinked the binary into "apps". The script is called from the Deskbar link, once it's set up it can be run either from there or in Terminal from $PATH.

I changed the settings folder as well, this will break current installs FYI.

humdingerb commented:
> Tried putting the binary into "apps" and it started segfaulting looking for its assets there,

Hmm. The current version 1.5.4-2 has only the binary and script in the apps folder, and the assets in the data folder. Weird that it should crash just because the binary isn't in a "DevilutionX" subfolder...

Peppersawce commented:
As far as I understood from the team, the "usual" install would have the binary in its own app folder with a single .mpq file containing all assets and the extra data paths supplied and the option to do that in code are meant for testing/dev purposes, mostly.

If all installs follow a similar convention I would assume either other people compiling it aren't facing the issue, the issue is ignored by the project because "you're not using standard paths" / "we don't support putting the binary anywhere else" / "not an issue"... or maybe the issue is on Haiku's side.

All in all, I find it weird as well, but as long as it works fine with reasonable workarounds I'm just going to ignore it, even if I wanted to do something about it I wouldn't know where to start. :)

humdingerb commented:
Fair enough. Having it in a all in a subfolder isn't a tragedy, just thought since v1.5.4-2 was already halfway there, with the assets being in $dataDir, it would be a minor change.
No matter. Carry on...

Peppersawce commented:
The assets are now packed inside an MPQ archive, this process depends on smpq so I'm making this a draft until the PR for it is merged.
![screenshot1](https://github.com/user-attachments/assets/13f2b4e8-1dde-462d-b12a-3db1f1ada830)
The newer hpkg is bigger than the older one btw. :)

Peppersawce commented:
Added a package for Diablo's shareware data, all the needed setup for it is done by running devilutionx.sh once.
